File: Round2/Blessing in Disguise/code.cpp
#include <bits/stdc++.h>

using namespace std;

string ltrim(const string &);
string rtrim(const string &);

/*
 * Complete the 'BubblyMaths' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER n as parameter.
 */

int BubblyMaths(int N) {
    // Write your code here below
    if (N == 1) return 1;
    if (N == 2) return 2;
    if (N == 3) return 6;
    if (N == 4) return 7;
    if (N % 4 == 1) return N + 2;
    if (N % 4 == 2) return N + 2;
    if (N % 4 == 3) return N - 1;
    return N + 1;

}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string n_temp;
    getline(cin, n_temp);

    int n = stoi(ltrim(rtrim(n_temp)));

    int result = BubblyMaths(n);

    fout << result << "\n";

    fout.close();

    return 0;
}

string ltrim(const string &str) {
    string s(str);

    s.erase(
        s.begin(),
        find_if(s.begin(), s.end(), not1(ptr_fun<int, int>(isspace)))
    );

    return s;
}

string rtrim(const string &str) {
    string s(str);

    s.erase(
        find_if(s.rbegin(), s.rend(), not1(ptr_fun<int, int>(isspace))).base(),
        s.end()
    );

    return s;
}

File: Round2/Blessing in Disguise/code.py
def BubblyMaths(N):
    if N == 1:
        return 1
    if N == 2:
        return 2
    if N == 3:
        return 6
    if N == 4:
        return 7
    if N % 4 == 1:
        return N + 2
    if N % 4 == 2:
        return N + 2
    if N % 4 == 3:
        return N - 1
    return N + 1

if __name__ == "__main__":
    n = int(input().strip())
    result = BubblyMaths(n)
    print(result)
